Police-Data-Accessibility-Project/data-sources-app-v2
# Create `Recent Searches` logic

### Fixes

* https://github.com/Police-Data-Accessibility-Project/data-sources-app/issues/468

### Description

* Update `/search` `GET` endpoint call to log recent searches
* Create `/user/recent-searches` `GET` endpoint for retrieving recent searches for user
* Update existing tests to validate the above behavior.

### Testing

* Run tests in `/tests` directory to confirm all run as expected.
* Inspect API documentation to confirm documentation is as expected.

### Performance

* With every search call, an additional database client call is now made to store a record of a given search
* Additional test overhead.

### Docs

* API documentation has been updated.

### Breaking Changes

* No breaking changes.
